common/src/main/java/com/microsoft/identity/common/internal/broker/ipc/IpcStrategyWithBackup.kt

```diff
@@ -0,0 +1,62 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.broker.ipc
+
+import android.os.Bundle
+import com.microsoft.identity.common.java.opentelemetry.AttributeName
+import com.microsoft.identity.common.java.opentelemetry.SpanExtension
+
+/**
+ * An IPC Strategy wrapper with backups.
+ * If the primary one fails, the backups will be used "best-effort".
+ * If all mechanisms fail, only the exception of the main one will be thrown.
+ * */
+class IpcStrategyWithBackup (
+    private val primary: IIpcStrategy,
+    private val backup: List<IIpcStrategy>): IIpcStrategy{
+
+    override fun communicateToBroker(bundle: BrokerOperationBundle): Bundle? {
+        return try {
+            primary.communicateToBroker(bundle)
+        } catch (t: Throwable) {
+            for(ipc in backup) {
+                try {
+                    val result = ipc.communicateToBroker(bundle)
+                    SpanExtension.current().setAttribute(
```

Comment:
@shahzaibj I'm planning to query # of success flows that use primary vs backup

Does this make sense?